python/GafferUI/PlugValueWidget.py

```diff
@@ -379,39 +393,39 @@ def _popupMenuDefinition( self ) :
 
 			menuDefinition.append(
 				"/Paste Value", {
-					"command" : functools.partial( Gaffer.WeakMethod( self.__setValue ), pasteValue ),
+					"command" : functools.partial( Gaffer.WeakMethod( self.__setValues ), pasteValue ),
 					"active" : self._editable() and pasteValue is not None
 				}
 			)
 
 			menuDefinition.append( "/CopyPasteDivider", { "divider" : True } )
 
-		if self.getPlug().getInput() is not None :
-			menuDefinition.append( "/Edit input...", { "command" : Gaffer.WeakMethod( self.__editInput ) } )
+		if any( p.getInput() is not None for p in self.getPlugs() ) :
+			menuDefinition.append( "/Edit input...", { "command" : Gaffer.WeakMethod( self.__editInputs ) } )
 			menuDefinition.append( "/EditInputDivider", { "divider" : True } )
 			menuDefinition.append(
 				"/Remove input", {
-					"command" : Gaffer.WeakMethod( self.__removeInput ),
-					"active" : self.getPlug().acceptsInput( None ) and not self.getReadOnly() and not Gaffer.MetadataAlgo.readOnly( self.getPlug() ),
+					"command" : Gaffer.WeakMethod( self.__removeInputs ),
+					"active" : not self.getReadOnly() and all( p.acceptsInput( None ) and not Gaffer.MetadataAlgo.readOnly( p ) for p in self.getPlugs() ),
```

Comment:
I'm not sure. I think it may be equally confusing to find you are allowed to perform _some_ operations on a widget which isn't editable, but not _all_. And from the widget you don't have a complete enough breakdown of what is read only and what isn't to really understand the reasons for what you're seeing. I'm inclined to leave as-is and wait for feedback from the real world (I'd be a bit surprised if any came up, at least for our initial use cases).

Reply:
Sounds like a plan. I guess if you wanted to remove all inputs from all exposures. It'd be nice if you could just select a load of lights and do it. If one was locked for some reason, even though it had no input. You'd have to go on a witch hunt to work out which light you had to deselect...

Reply:
Even with that change, you'll still end up persecuting witches in the case that a connected plug is read only. I'm sure part of my reluctance is a desire to get away from coding UIs and to get this wrapped up before sprog day, but this seems like enough of a corner case that we could leave it.

Reply:
Haha, yeah, its totally fine :)